src/backend/distributed/executor/multi_utility.c

```diff
@@ -3193,6 +3218,81 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 }
 
 
+/*
+ * SetSearchPathToCurrentSearchPathCommand generates a command which can
+ * set the search path to the exact same search path that the issueing node
+ * has.
+ *
+ * If the current search path is null (or doesn't have any valid schemas),
+ * the function returns NULL.
+ */
+static char *
+SetSearchPathToCurrentSearchPathCommand(void)
+{
+	StringInfo setCommand = NULL;
+	char *currentSearchPath = CurrentSearchPath();
+
+	if (currentSearchPath)
+	{
+		setCommand = makeStringInfo();
+		appendStringInfo(setCommand, "SET search_path TO %s;", currentSearchPath);
+		return setCommand->data;
+	}
+
+	return NULL;
+}
+
+
+/*
+ * CurrentSearchPath is a C interface for calling current_schemas(bool) that
+ * PostgreSQL exports.
+ *
+ * CurrentSchemas returns all the schemas in the seach_path that are seperated
+ * with comma (,) sign. The returned string can be used to set the search_path.
+ *
+ * The function omits implicit schemas.
+ *
+ * The function returns NULL if there are no valid schemas in the search_path,
+ * mimicing current_schemas(false) function.
+ */
+static char *
+CurrentSearchPath(void)
+{
+	StringInfo currentSearchPath = makeStringInfo();
+	List *searchPathList = fetch_search_path(false);
+	ListCell *searchPathCell;
+	bool schemaAdded = false;
+
+	foreach(searchPathCell, searchPathList)
+	{
+		char *schemaName = NULL;
+
+		if (schemaAdded)
+		{
+			appendStringInfoString(currentSearchPath, ",");
+		}
+
+		/* watch out for deleted namespace */
+		schemaName = get_namespace_name(lfirst_oid(searchPathCell));
+		if (schemaName)
+		{
+			appendStringInfoString(currentSearchPath, quote_identifier(schemaName));
+
+			schemaAdded = true;
+		}
+		else
```

Comment:
This makes it a lot less readable. 

How about
```c
	/* watch out for deleted namespace */
	schemaName = get_namespace_name(lfirst_oid(searchPathCell));
	if (schemaName)
	{
		if (schemaAdded)
		{
			appendStringInfoString(currentSearchPath, ",");
		}
		
		appendStringInfoString(currentSearchPath, quote_identifier(schemaName));
		schemaAdded = true;
	}
```

Reply:
Yes, this is much better. I'll update the code accordingly

Reply:
Please see my new code-style now